### PR TITLE
Fix the inline asm code build error with greenhills compiler

### PR DESCRIPTION
**Pull request by extinguish**
## Summary
Fixed the compilation error caused by incompatible inline assembly code for the Green Hills compiler platform.

### CODE REVIEW

File: arch/arm/include/syscall.h [hunk 1]
```diff
@@ -136,6 +136,18 @@
 
 static inline uintptr_t sys_call0(unsigned int nbr)
 {
+#ifdef __ghs__
```
Comment (anchao) [inline arch/arm/include/syscall.h:139]: have you enable "-gnu99" and "--gnu_asm" in command line ? I think you would better to upstream the GHS toolchain support first

Comment (extinguish) [reply arch/arm/include/syscall.h:139]: yes, the "-gnu99" and "--gnu_asm" is enabled in another pull request

Comment (extinguish) [reply arch/arm/include/syscall.h:139]: > have you enable "-gnu99" and "--gnu_asm" in command line ? I think you would better to upstream the GHS toolchain support first

yes, the "-gnu99" and "--gnu_asm" is enabled in another pull request

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/arm/include/syscall.h (hunk 1 above)
Comment (anchao) [reply arch/arm/include/syscall.h:139]: > > have you enable "-gnu99" and "--gnu_asm" in command line ? I think you would better to upstream the GHS toolchain support first
> 
> yes, the "-gnu99" and "--gnu_asm" is enabled in another pull request

which pull request?

Comment (extinguish) [reply arch/arm/include/syscall.h:139]: > > > have you enable "-gnu99" and "--gnu_asm" in command line ? I think you would better to upstream the GHS toolchain support first
> > 
> > 
> > yes, the "-gnu99" and "--gnu_asm" is enabled in another pull request
> 
> which pull request?

is in https://github.com/apache/nuttx/pull/12877 new pull request

**Pull request closed without merging**